scrapy_poet/injection.py

```diff
@@ -387,11 +402,16 @@ def is_callback_requiring_scrapy_response(
         # Let's assume response is going to be used.
         return True
 
-    if first_parameter.annotation is first_parameter.empty:
+    callback_type_hints = get_type_hints(callback)
+    first_parameter_type_hint = callback_type_hints.get(first_parameter_key, _UNDEFINED)
+    if first_parameter_type_hint is _UNDEFINED:
+        return True
+
+    if first_parameter_type_hint is first_parameter.empty:
         # There's no type annotation, so we're probably using response here.
```

Comment:
This comment belongs to the code path above as well, right? Would it make sense to extract the logic to a function or a method?

Reply:
Actually, I think we can remove this second if statement altogether.